### PR TITLE
US104726 - Support page size in query param of demo

### PR DESCRIPTION
**Pull request by mtjandra**
pagesize

This is step one:
* Allow configuring default page size mock server uses with query parameter

### CODE REVIEW

File: demo/d2l-evaluation-hub/d2l-evaluation-hub-activities-list.html [hunk 1]
```diff
@@ -21,7 +21,9 @@
 			import mappingsParser from './demo-data-parser';
 			import data from './table';
 
-			const mappings = mappingsParser(data);
+			const url = new URL(window.location);
```
Comment (svanherk) [inline demo/d2l-evaluation-hub/d2l-evaluation-hub-activities-list.html]: This is making the IE11 problem worse.  But we should know in the next two weeks if we have to continue supporting it.

Comment (mtjandra) [reply demo/d2l-evaluation-hub/d2l-evaluation-hub-activities-list.html]: This is for demo data only, and demo data has polyfill. Also IE demo doesn't work at all right now, I'm told this is a known issue

**Comment by jmurray87** [comment]
Won't affect the change we made for the filter options if its just in the demo,  the change I made still respects the pageSize query param, but sets it to max int if its not present

**Comment by mtjandra** [comment]
> Won't affect the change we made for the filter options if its just in the demo, the change I made still respects the pageSize query param, but sets it to max int if its not present

That makes the default use case not ideal, since you would no longer see load more. This is for demo purposes only

**Comment by jmurray87** [comment]
The change I made only applies to the filters/{guid} route

File: demo/d2l-evaluation-hub/d2l-evaluation-hub-activities-list.html (hunk 1 above)
Comment (svanherk) [reply demo/d2l-evaluation-hub/d2l-evaluation-hub-activities-list.html]: Yup, it's currently broken and tracked in a defect, but this is just making things worse (if prod ever was fixed, we'd want our demos to work.  There's a polyfill, but that's just hiding problems and should be removed).

**Comment by svanherk** [comment]
> The change I made only applies to the filters/{guid} route

But won't the query param be passed down to that route?  Like if the my-unaccessed-activities route has a `pageSize`, does the filter link also have that? (this is what I was unsure about, it may be totally fine)

File: demo/d2l-evaluation-hub/d2l-evaluation-hub-activities-list.html (hunk 1 above)
Comment (mtjandra) [reply demo/d2l-evaluation-hub/d2l-evaluation-hub-activities-list.html]: Yes IE and fixing filtering demo data should be handled seperatley from what I'm doing here. Thanks for the heads up!

**Comment by mtjandra** [comment]
> But won't the query param be passed down to that route? Like if the my-unaccessed-activities route has a pageSize, does the filter link also have that? (this is what I was unsure about, it may be totally fin

Good call out guys. I will definitely be thinking of this once it comes to the implementation!

File: demo/d2l-evaluation-hub/demo-data-parser.js [hunk 2]
```diff
@@ -349,7 +349,11 @@ function getHrefForMasterTeacher(id) {
 *
 * `mappings` (which is the return value) maps urls to siren endpoints to be consumed by the interceptor
 */
-function getMappings(table) {
+function getMappings(table, pageSize) {
+	if (!pageSize) {
```
Comment (AlexBedley) [inline demo/d2l-evaluation-hub/demo-data-parser.js]: IMO we should make the second param to `getMappings` something like `options` and `pageSize` should be a member of that object. Just so when someone adds another config thing they don't add another parameter

File: demo/d2l-evaluation-hub/d2l-evaluation-hub.html [hunk 3]
```diff
@@ -21,7 +21,9 @@
 			import mappingsParser from './demo-data-parser';
 			import data from './table';
 
-			const mappings = mappingsParser(data);
+			const url = new URL(window.location);
+			const pageSize = url.searchParams.get('pageSize') || 3;
```
Comment (AlexBedley) [inline demo/d2l-evaluation-hub/d2l-evaluation-hub.html]: I find it weird that we have 2 places with a default value. I would remove one of them

Comment (mtjandra) [reply demo/d2l-evaluation-hub/d2l-evaluation-hub.html]: Alright updated when fixing merge conflicts

File: demo/d2l-evaluation-hub/demo-data-parser.js (hunk 2 above)
Comment (mtjandra) [reply demo/d2l-evaluation-hub/demo-data-parser.js]: Alright updated when fixing merge conflicts

**Comment by mtjandra** [comment]
> It doesn't handle the cases where you want a dynamic page size but it's better than it was before! 

Correct, next PR will handle dynamic page sizes, this is step one

**Pull request merged**